### ISSUE TITLE
MAINT: Formalizing the `con` format

### ISSUE THREAD

**Issue by HaoZeke** [TheochemUI/eOn#175]
In general the `con` file format is great. However, there is a dependence on the Atom ID which is currently implicit, it determines the ordering of components, and therefore the order of counts and masses in the header as well.

There are a few options here:
- Store Atomic ID in the Matter object explicitly, and use its ordering (current behavior)
  - This is implicitly enforced but it is a bit awkward
  - Would make it easier to ensure consistency wrt LAMMPS if this is explicit
- Choose a default order, e.g. say the atomic mass order
  - Programmatically easier, but then mapping to say, LAMMPS might need more work

Basically for the first case, Matter needs some rework, so this is part of #170.

**Comment by HaoZeke** [comment]
Atom ID is now part of the Matter object. Documentation needed to close this out.

**Comment by HaoZeke** [comment]
https://lode-org.github.io/readcon-core/index.html
